client/performance-profiler/components/dashboard-content/index.tsx

```diff
@@ -49,7 +50,15 @@ export const PerformanceProfilerDashboardContent = ( {
 					tbt={ tbt }
 					history={ history }
 				/>
-				<NewsletterBanner link={ `/speed-test-tool/weekly-report?url=${ url }&hash=${ hash }` } />
+				<NewsletterBanner
+					link={ `/speed-test-tool/weekly-report?url=${ url }&hash=${ hash }` }
+					onClick={ () => {
+						recordTracksEvent( 'calypso_performance_profiler_weekly_report_cta_click', {
```

Comment:
Noob question: Does user tracking happen automatically inside `recordTracksEvent`?

Reply:
Great question! The `user_id` is included in the track events data sent by Calypso, as you can see in the `_ul` field [here](https://github.com/Automattic/wp-calypso/pull/94369#discussion_r1752245800). I think this is populated when initializing Calypso [here](https://github.com/Automattic/wp-calypso/blob/e533c35d5a4e8d0512ed100d3ccc0d8757fc9c82/packages/calypso-analytics/src/tracks.ts#L182)

More info PCYsg-4KT-p2.

Reply:
Yes, there's a default object that gets passed to each Tracks event. The user ID is set in the `_ui` property of each event:

https://github.com/Automattic/wp-calypso/blob/65465d241f774bf92f7ddcff3726bf117c9a6f88/packages/calypso-analytics/src/tracks.ts#L88-L90